Source/Core/VideoCommon/OnScreenUI.cpp

```diff
@@ -356,35 +356,37 @@ void OnScreenUI::SetScale(float backbuffer_scale)
 }
 void OnScreenUI::SetKeyMap(const DolphinKeyMap& key_map)
 {
-  // Right now this is a 1:1 mapping. But might not be true later
   static constexpr DolphinKeyMap dolphin_to_imgui_map = {
       ImGuiKey_Tab,       ImGuiKey_LeftArrow, ImGuiKey_RightArrow, ImGuiKey_UpArrow,
       ImGuiKey_DownArrow, ImGuiKey_PageUp,    ImGuiKey_PageDown,   ImGuiKey_Home,
       ImGuiKey_End,       ImGuiKey_Insert,    ImGuiKey_Delete,     ImGuiKey_Backspace,
-      ImGuiKey_Space,     ImGuiKey_Enter,     ImGuiKey_Escape,     ImGuiKey_KeyPadEnter,
+      ImGuiKey_Space,     ImGuiKey_Enter,     ImGuiKey_Escape,     ImGuiKey_KeypadEnter,
       ImGuiKey_A,         ImGuiKey_C,         ImGuiKey_V,          ImGuiKey_X,
       ImGuiKey_Y,         ImGuiKey_Z,
   };
-  static_assert(dolphin_to_imgui_map.size() == ImGuiKey_COUNT);  // Fail if ImGui adds keys
```

Comment:
Is this static assert no longer correct? `DolphinKey` is a `Common::EnumMap` which doesn't do bounds-checking (but it looks like the more relevant thing is that `DolphinKey::Z` is the last key in the map, and that's less likely to change).

Reply:
Yeah, it's no longer correct.  While the [key still exists](https://github.com/iwubcode/dolphin/blob/imgui_update/Externals/imgui/imgui.h#L1478), the number of entries is much much larger (compare with [original](https://github.com/ocornut/imgui/blob/v1.85/imgui.h#L1378C19-L1378C19)).  There was a big change to 1.87 inputs in imgui from what I understand.

My assumption was that this would be ok because both `key_map` and `dolphin_to_imgui_map` are the same types and are both bound by `DolphinKey::Z` as you mention.  If `DolphinKeyMap` changed both of these maps would need to change as well.